code/modules/cargo/packs/organic.dm

```diff
@@ -185,52 +181,58 @@
-				continue
+/datum/supply_pack/organic/pizza/fill(obj/structure/closet/crate/new_crate)
+	. = ..()
+	var/list/rng_pizza_list = pizza_types.Copy()
+	for(var/i in 1 to 5)
+		if(add_anomalous(new_crate))
+			continue
+		if(add_boombox(new_crate))
+			continue
+		add_normal_pizza(new_crate, rng_pizza_list)
 
-		if(!boombox_provided)
-			var/boombox_lottery = rand(1,boombox_tickets)
-			var/boombox_type
-			switch(boombox_lottery)
-				if(1 to 2)
-					boombox_type = /obj/item/pizzabox/bomb/armed //explodes after opening
-				if(3)
-					boombox_type = /obj/item/pizzabox/bomb //free bomb
+/// adds the chance for an infinite pizza box
+/datum/supply_pack/organic/pizza/proc/add_anomalous(obj/structure/closet/crate/new_crate)
+	if(anomalous_box_provided)
+		return FALSE
+	if(!prob(infinite_pizza_chance))
+		return FALSE
+	new /obj/item/pizzabox/infinite(new_crate)
+	anomalous_box_provided = TRUE
+	log_game("An anomalous pizza box was provided in a pizza crate at during cargo delivery.")
+	if(prob(50))
+		addtimer(CALLBACK(src, PROC_REF(anomalous_pizza_report)), rand(30 SECONDS, 180 SECONDS))
+		message_admins("An anomalous pizza box was provided in a pizza crate at during cargo delivery.")
+	else
+		message_admins("An anomalous pizza box was silently created with no command report in a pizza crate delivery.")
+	return TRUE
 
-			if(boombox_type)
-				new boombox_type(C)
-				qdel(P)
-				boombox_provided = TRUE
-				log_game("A pizza box bomb was created by a pizza crate delivery.")
-				message_admins("A pizza box bomb has arrived in a pizza crate delivery.")
-				continue
+/// adds a chance of a pizza bomb replacing a pizza
+/datum/supply_pack/organic/pizza/proc/add_boombox(obj/structure/closet/crate/new_crate)
+	if(boombox_provided)
+		return FALSE
+	if(!prob(bomb_pizza_chance))
+		return FALSE
+	var/boombox_type = (prob(bomb_dud_chance)) ? /obj/item/pizzabox/bomb : /obj/item/pizzabox/bomb/armed
+	new boombox_type(new_crate)
+	boombox_provided = TRUE
+	log_game("A pizza box bomb was created by a pizza crate delivery.")
+	message_admins("A pizza box bomb has arrived in a pizza crate delivery.")
+	return TRUE
 
-		//here we randomly replace our pizzas for a chance at the full range
-		var/obj/item/food/pizza/replacement_type = pick_weight(pizza_types)
-		pizza_types -= replacement_type
-		if(replacement_type && !istype(P.pizza, replacement_type))
-			QDEL_NULL(P.pizza)
-			P.pizza = new replacement_type
-			P.boxtag = P.pizza.boxtag
-			P.boxtag_set = TRUE
-			P.update_appearance()
+/// adds a randomized pizza from the pizza list
+/datum/supply_pack/organic/pizza/proc/add_normal_pizza(obj/structure/closet/crate/new_crate, list/rng_pizza_list)
+	var/randomize_pizza = pick_weight(rng_pizza_list)
+	rng_pizza_list -= randomize_pizza
+	var/obj/item/pizzabox/new_pizza_box = new(new_crate)
+	new_pizza_box.pizza = new randomize_pizza
+	new_pizza_box.boxtag = new_pizza_box.pizza.boxtag
+	new_pizza_box.boxtag_set = TRUE
+	new_pizza_box.update_appearance()
```

Comment:
What actually needs to update here?

Reply:
the sprite doesnt start with the little text on it, so after it has a pizza in it and it knows what to write on the box it needs to update into having writing on the box

Reply:
i changed it to update the icon and the description because otherwise it has the sprite with writing but doesnt tell you what the writing is.